src/deluge/gui/ui/sample_marker_editor.cpp

```diff
@@ -142,10 +145,28 @@ void SampleMarkerEditor::writeValue(uint32_t value, MarkerType markerTypeNow) {
 		getCurrentSampleHolder()->startPos = value;
 	}
 	else if (markerTypeNow == MarkerType::LOOP_START) {
-		getCurrentMultisampleRange()->sampleHolder.loopStartPos = value;
+		if (loopLocked) {
+			int intendedLoopEndPos = value + loopLength;
+			if (intendedLoopEndPos < getCurrentSampleHolder()->endPos) {
+				getCurrentMultisampleRange()->sampleHolder.loopStartPos = value;
+				getCurrentMultisampleRange()->sampleHolder.loopEndPos = intendedLoopEndPos;
+			}
+		}
+		else {
+			getCurrentMultisampleRange()->sampleHolder.loopStartPos = value;
+		}
 	}
 	else if (markerTypeNow == MarkerType::LOOP_END) {
-		getCurrentMultisampleRange()->sampleHolder.loopEndPos = value;
+		if (loopLocked) {
+			int intendedLoopStartPos = value - loopLength;
+			if (intendedLoopStartPos > getCurrentSampleHolder()->startPos) {
```

Comment:
Same here

Reply:
Addressed in https://github.com/SynthstromAudible/DelugeFirmware/pull/293/commits/1b6dc94235e14b644c91bfe5d96965dab55fe28e